src/index.ts

```diff
@@ -172,32 +624,133 @@ export async function isSupported(): Promise<boolean> {
+
+    if (exchangeKeyPair && writeKeyPair) {
+      await components.storage.setItem("exchange-key", exchangeKeyPair)
+      await components.storage.setItem("write-key", writeKeyPair)
+    }
+
+    bwDeleteDatabase(keystoreDB)
+  }
+
+  const localforageDB = await bwOpenDatabase("localforage")
+
+  if (localforageDB) {
+    const accountUcan = await bwGetValue(localforageDB, "keyvaluepairs", "ucan")
+    const permissionedUcans = await bwGetValue(localforageDB, "keyvaluepairs", "webnative.auth_ucans")
+    const rootKey = await bwGetValue(localforageDB, "keyvaluepairs", "readKey")
+    const authedUser = await bwGetValue(localforageDB, "keyvaluepairs", "webnative.auth_username")
+
+    if (rootKey && isString(rootKey)) {
+      const anyUcan = accountUcan || (Array.isArray(permissionedUcans) ? permissionedUcans[ 0 ] : undefined)
+      const accountDID = anyUcan ? Ucan.rootIssuer(anyUcan) : null
+      if (!accountDID) throw new Error("Failed to retrieve account DID")
 
-export * as dns from "./dns/index.js"
-export * as ipfs from "./ipfs/index.js"
-export * as keystore from "./keystore.js"
-export * as machinery from "./common/index.js"
-export * as crypto from "./crypto/index.js"
+      await RootKey.store({
+        accountDID,
+        crypto: components.crypto,
+        readKey: Uint8arrays.fromString(rootKey, "base64pad"),
+      })
+    }
+
+    if (accountUcan) {
+      await components.storage.setItem(
+        components.storage.KEYS.ACCOUNT_UCAN,
+        accountUcan
+      )
+    }
+
+    if (authedUser) {
+      await components.storage.setItem(
+        components.storage.KEYS.SESSION,
+        JSON.stringify({
+          type: isConfidentialAuthConfiguration(config) ? CONFIDENCES_SESSION_TYPE : WEB_CRYPTO_SESSION_TYPE,
+          username: authedUser
+        })
+      )
+    }
+
+    bwDeleteDatabase(localforageDB)
+  }
+}
+
+
+function bwDeleteDatabase(db: IDBDatabase): void {
+  const name = db.name
+  db.close()
+  indexedDB.deleteDatabase(name)
+}
+
+
+function bwGetValue(db: IDBDatabase, storeName: string, key: string): Promise<Maybe<unknown>> {
+  return new Promise((resolve, reject) => {
+    if (!db.objectStoreNames.contains(storeName)) return resolve(null)
+
+    const transaction = db.transaction([ storeName ], "readonly")
+    const store = transaction.objectStore(storeName)
+    const req = store.get(key)
+
+    req.onerror = () => {
+      // No store, moving on.
+      resolve(null)
+    }
+
+    req.onsuccess = () => {
+      resolve(req.result)
+    }
+  })
+}
+
+
+function bwOpenDatabase(name: string): Promise<Maybe<IDBDatabase>> {
+  return new Promise((resolve, reject) => {
+    const req = self.indexedDB.open(name)
+
+    req.onerror = () => {
+      // No database, moving on.
+      resolve(null)
+    }
+
+    req.onsuccess = () => {
+      resolve(req.result)
+    }
+  })
+}
+
+
+
+// 🛠
+
+
+export function extractConfig(opts: Partial<Components> & Configuration): Configuration {
```

Comment:
Ah yes, that should fail because `tag` is a required parameter.
I just omitted it in the readme.

I wasn't sure if it should infer it from permissions.
I think I prefer explicitness here.
I'll add the `tag` to the example.

Reply:
Sounds good, I'm with explicit. 💯